File: 02-spring-boot-data/mongodb/src/main/java/vip/yazilim/springboot/basics/mongodb/service/IPersonService.java
package vip.yazilim.springboot.basics.mongodb.service;

import java.util.List;
import vip.yazilim.springboot.basics.mongodb.entity.Person;

/**
 *
 * @author Emre Sen
 * @date Jan 23, 2019
 * @contact maemresen07@gmail.com
 */

public interface IPersonService {

    // ...
    Person insert(Person person);

    Person update(Person person);

    void delete(int id);

    // ...
    List<Person> findAll();

    Person findById(int id);

    Person findFirstByNameAsc();
    
    List<Person> findByName(String name);
    
    List<Person> findByLocation(String location);

}

File: 03-spring-boot-thymeleaf/02-thymeleaf-bootstrap-webjars/src/main/kotlin/vip/yazilim/thymeleaf/bootstrap_webjars/service/IPersonService.kt
package vip.yazilim.thymeleaf.bootstrap_webjars.service

import vip.yazilim.thymeleaf.bootstrap_webjars.model.Person

/**
 *
 * @author maemresen - maemresen@yazilim.vip
 * 27.12.2020
 */
interface IPersonService {
    fun getAll(): List<Person>
    fun getByName(name: String): Person
}